concurrency/_index.md

```diff
@@ -1,7 +1,170 @@
+The caller can poll this `Future` object and then perform some action using the resulting object once the task has completed. With a `CompletableFuture` or `CompletionStage`, the caller can chain dependent actions to the completion of the task rather than polling.
+
+A task is developed by creating a class that must implement `java.util.concurrent.Callable` or `java.lang.Runnable`, or any of the functional interfaces that can be supplied to a `java.util.concurrent.CompletionStage`.
+A Callable task class can optionally return a result to the caller. When tasks are submitted to a managed executor or managed `CompletionStage`, they become contextual, behaving in the same manner as they had within the original container.
+This means that the task runs under the same contextual information and state that was present on the thread that submits the task to the `ManagedExecutorService`.
+For example, a session may contain the username of an authenticated user. A task that is submitted to a `ManagedExecutorService` while running as that user, will be run by the `ManagedExecutorService` as that user. 
+A task class that implements `Runnable` may be structured as follows:
+
+```Java
+public class MyTask implements Runnable {
+    // Inject resources, if needed
+
+    public void run() {
+        // execute logic
+    }
+}
+```
+
+For more fine grained control, a task class can optionally implement the `jakarta.enterprise.concurrent.ManagedTask` interface, which enables the ability to perform actions upon lifecycle events.
+This is achieved by registering a `jakarta.enterprise.concurrent.ManagedTaskListener` instance with the task. A use case for implementing `ManagedTask` may be to initiate a process when a particular lifecycle event occurs.
+
+To pass a task to a `ManagedExecutorService`, the `ManagedExecutorService` can be obtained for use within a resource through use of the `Java Naming and Directory Interface (JNDI)`.
+The `ManagedExecutorService` can be configured by the application via a `ManagedExecutorDefinition` annotation or via XML within the `web.xml` or `application.xml` deployment descriptors.
+Alternatively, the default `java:comp/DefaultManagedExecutorService` can also be used.
+
+After the resource has been obtained, the task can be submitted to it by calling upon any one of the `submit()`, `execute()`, `invokeAll()`, `invokeAny()`, `runAsync()`, or `supplyAsync()` methods.
+The following code demonstrates how to obtain the `ManagedExecutorService` resource and submit a task to it.
+
+```Java
+@Resource(lookup = "java:comp/DefaultManagedExecutorService")
+ManagedExecutorService managedExecutorService;
+
+// Pass task class to the ManagedExecutorService
+Future<ResultType> futureObject = managedExecutorService.submit(myTask);
+```
+
+If there are a number of tasks which need to be executed in parallel, it is possible to do so by passing a list of `Runnable` or `Callable` tasks to the `ManagedExecutorService` `invokeAll()` method.
```

Comment:
The invokeAll API method only accepts Callable.
```suggestion
If there are a number of tasks which need to be executed in parallel, it is possible to do so by passing a list of `Callable` tasks to the `ManagedExecutorService` `invokeAll()` method.
```

Reply:
Good catch. I must have missed this. I've updated it in my patch.